data_structures/binary_trees/BT implementation.c

```diff
@@ -0,0 +1,92 @@
+//Binary Tree implementation---> creation,insertion and deletion , traversal,searching
+
+#include<stdio.h>
+#include<stdlib.h>
+struct node
+{
+	int info;
+	struct node* left;
+	struct node* right;
+};
+struct node* getnode()
+{
+	struct node* p;
+	p=(struct node*)malloc(sizeof(struct node));
+	return p;
+}
+struct node* maketree(int x)
+{
+	struct node* p;
+	p=getnode();
+	p->info=x;
+	p->left=NULL;
+	p->right=NULL;	
+	return p;
+}
+struct node* setleft(struct node* p, int x)
+{
+
+	if(p==NULL)
+	printf("Void insrtion");
+	else if(p->left!=NULL)
+	printf("Invalid insertion");
+	else 
+	p->left=maketree(x);
+}
+struct node* setright(struct node* p, int x)
+{
+
+	if(p==NULL)
+	printf("Void insrtion");
+	else if(p->right!=NULL)
+	printf("Invalid insertion");
+	else 
+	p->right=maketree(x);
+}
+void pretrav(struct node* p)
+{int count =0;
+		if(p!=NULL)
+	{
+	pretrav(p->left);
+	pretrav(p->right);
+	printf("%d",p->info);
+		
+	
+}
+
+}
+void posttrav(struct node* p)
+{
+		if(p!=NULL)
+	{
+	printf("%d",p->info);
+	posttrav(p->left);
+	posttrav(p->right);

```

Comment:
In postorder traversal, we first traverse to the left sub tree, then traverse the right sub tree, & then print the data in the current node.